internal/services/storage/storage_account_data_plane_helpers.go

```diff
@@ -64,8 +66,10 @@ func waitForDataPlaneToBecomeAvailableForAccount(ctx context.Context, client *cl
 			return fmt.Errorf("building Blob Service Poller: %+v", err)
 		}
 		poller := pollers.NewPoller(pollerType, initialDelayDuration, pollers.DefaultNumberOfDroppedConnectionsToAllow)
-		if err := poller.PollUntilDone(ctx); err != nil {
-			return fmt.Errorf("waiting for the Blob Service to become available: %+v", err)
+		if err = poller.PollUntilDone(ctx); err != nil {
+			if !connectionError(err) {
+				return fmt.Errorf("waiting for the Blob Service to become available: %+v", err)
+			}
```

Comment:
Clarifying my understanding here, if we get a connection error while polling we silently ignore it?

Reply:
Correct, whilst this is counter to typical behaviour in the provider it is necessary in this unique case until 5.0 and we complete the separation of Data Plane APIs from all Resource Manager resources.